src/main/java/io/split/android/client/validators/FlagSetsValidatorImpl.java

```diff
@@ -0,0 +1,47 @@
+package io.split.android.client.validators;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.TreeSet;
+
+import io.split.android.client.utils.logger.Logger;
+
+public class FlagSetsValidatorImpl implements SplitFilterValidator {
+
+    private static final String FLAG_SET_REGEX = "^[a-z][_a-z0-9]{0,49}$";
+
+    /**
+     * Validates the flag sets and returns a list of
+     * de-duplicated and alphanumerically ordered valid flag sets.
+     *
+     * @param values list of flag sets
+     * @return list of unique alphanumerically ordered valid flag sets
+     */
+    @Override
+    public List<String> cleanup(List<String> values) {
+        if (values == null || values.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        TreeSet<String> cleanedUpSets = new TreeSet<>();
+        for (String set : values) {
+            if (set == null || set.isEmpty()) {
+                continue;
+            }
+
+            if (set.startsWith(" ") || set.endsWith(" ")) {
```

Comment:
I think it is better to compare the length of the original and trimmed versions of the string to log the warning, in case the trim operation removes other white space characters like tabulation, etc.